jekyll/_cci2/server-3-operator-externalizing-services.adoc

```diff
@@ -0,0 +1,84 @@
+---
+version:
+- Server v3.x
+- Server Admin
+---
+= CircleCI Server v3.x Configuring External Services
+:page-layout: classic-docs
+:page-liquid:
+:icons: font
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== External Database and Vault Setup for CircleCI Installations
+This document describes how to configure the following external services for use with CircleCI Server 3.0 Installation:
+
+* PostgreSQL v12.6
+
+== PostgreSQL
+
+=== Best Practices for your PostgreSQL
+[width="60%",frame="topbot",options="header,footer"]
+Consider running at least two or more PostgreSQL replicas to enable recovery from primary failure and for backups. Following are the recommended specifications of the PostgreSQL machines:
+
+|=====================================================================================
+|# of Daily Active Users| # of PostgreSQL Replicas | CPU   | RAM  | Disk  | NIC Speed 
+|<50                    |2                         |8 Cores|16 GB |100 GB | 1 Gbps
+|50 - 250               |2                         |8 Cores|16 GB |200 GB | 1 Gbps
+|250 - 1000             |3                         |8 Cores|32 GB |500 GB | 10 Gbps
+|1000 - 5000            |3                         |8 Cores|32 GB |1 TB   | 10 Gbps
+|5000+                  |3                         |8 Cores|32 GB |1 TB   | 10 Gbps
+|=====================================================================================
+
+=== Preparing the Database
+If you are configuring an externalized PostgreSQL instance as part of a new CircleCI 3.0 install, it is recommended that you login to your PostgreSQL instance and run the following commands. If you plan on using a different method, you must ensure that the databases and extensions outlined below are configured prior to running CircleCI Server 3.0 for the first time.
```

Comment:
Since we have the capacity to do this, shouldn't this be executed automatically in server? Do we have to continuously update these commands every time a new service is onboarded and nudge customers to run migrations?

Reply:
> Since we have the capacity to do this, shouldn't this be executed automatically in server? Do we have to continuously update these commands every time a new service is onboarded and nudge customers to run migrations?

Heck yeah! I am always for automating as many things as we can.  TBH I am not sure how to accomplish this without doing some homework.  If this is trivial then please let me know how and I will get I done. 

Otherwise, we can make a card for this.

Reply:
Did anything come of this? Just wondering if a card can be made so the thought isn't lost once this PR gets merged :-)

Reply:
@nsmith5 ^^